### PR TITLE
[PT] Translation of Ecto/Changesets

### PR DESCRIPTION
**Pull request by pedrobarbiero**

### CODE REVIEW

File: pt/lessons/ecto/changesets.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
+---
+version: 1.0.0
+title: Changesets
+---
+
+Para inserir, atualizar ou excluir as informações de um banco de dados, `Ecto.Repo.insert/2`, `update/2` e `delete/2` é necessário um changeset como primeiro parâmetro.
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: I believe "...requerem um changeset..." works better.

File: pt/lessons/ecto/changesets.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+Mas o que são exatamente changesets?
+
+Uma tarefa comum para quase todos os desenvolvedores é verificar os dados de entrada para possíveis erros — Queremos ter certeza de que os dados estão no estado correto antes de tentarmos usá-los para nossos propósitos.
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: "...dados de entrada **por** possíveis erros - **queremos**..."

Just two small corrections here.

File: pt/lessons/ecto/changesets.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+defmodule User do
+  use Ecto.Schema
+
+  schema "users" do
+    field(:name, :string)
+  end
+end
+```
+
+Para criar um changeset usando o schema `User`, vamos usar `Ecto.Changetset.cast/4`:
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: There is a typo here in Changeset: `Ecto.Changetset.cast/4`

File: pt/lessons/ecto/changesets.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+```elixir
+iex> Ecto.Changeset.cast(%User{name: "Bob"}, %{}, [:name])
+#Ecto.Changeset<action: nil, changes: %{}, errors: [], data: #User<>,
+ valid?: true>
+ ```
+
+O primeiro parâmetro é o dado original - uma stuct `%User{}` vazia neste caso. 
+Ecto é inteligente o suficiente para encontrar o schema baseado na própria estrutura.
+O segundo parâmetro são as alterações que queremos fazer - apenas uma map vazio.
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: I believe we have been using masculine genre for maps throughout the lessons: "... apenas **um** map vazio".

File: pt/lessons/ecto/changesets.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  action: nil,
+  changes: %{},
+  errors: [name: {"can't be blank", [validation: :required]}],
+  data: #User<>,
+  valid?: false
+>
+```
+
+Caso você tente usar `Repo.insert(changeset)` com o changeset descrito acima, irá receber um `{:error, changeset}` de volta com o mesmo erro, então você não precisa checar `changeset.valid?` você mesmo toda vez.
+É mais facil tentar executar o insert, update ou delete, então processar os erros depois, caso existam.
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: Typo: **fácil**

Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: "...delete, **e** então..."

File: pt/lessons/ecto/changesets.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+  errors: [
+    name: {"should be at least %{count} character(s)",
+     [count: 2, validation: :length, min: 2]}
+  ],
+  data: #User<>,
+  valid?: false
+>
+```
+
+Você pode se surpreender já que a mensagem de erro contém o `%{count}` enigmático - isto é para ajudar a tradução para outras línguas; se você quiser exibir os erros diretamente para o usuário, você pode torná-los legíveis usando [`traverse_errors/2`](https://hexdocs.pm/ecto/Ecto.Changeset.html#traverse_errors/2) - Dê uma olhada no exemplo fornecido na documentação.
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: Small typo at the end of the phrase: "... - **dê** uma ..."

File: pt/lessons/ecto/changesets.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    changeset
+  else
+    add_error(changeset, :name, "is not a superhero")
+  end
+end
+```
+
+Acima nós introduzimos duas novas funções auxiliares: [`get_field/3`](https://hexdocs.pm/ecto/Ecto.Changeset.html#get_field/3) e [`add_error/4`](https://hexdocs.pm/ecto/Ecto.Changeset.html#add_error/4). O que elas fazem é quase auto-explicativo, mas eu aconselho você a verificar os links da documentação.
+
+É uma boa pratica retornar sempre um `%Ecto.Changeset{}`, então você pode usar o perador `|>` e facilitar a adição de mais validações posteriormente:
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: Small typo: "... pode usar **o operador**..."

File: pt/lessons/ecto/changesets.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
+---
+version: 1.0.0
```
Comment (anderkonzen) [inline pt/lessons/ecto/changesets.md]: I made the review against the 1.1.1 English version, so I believe we can use this version here to keep it in sync.

**Comment by pedrobarbiero** [comment]
@anderkonzen thank you for comments.
Ready to re-check 😄

**Comment by anderkonzen** [comment]
Thanks @pedrobarbiero ! Everything is alright!

When travis is fixed we should be able to merge.

**Comment by doomspork** [comment]
Thank you @pedrobarbiero and @anderkonzen! Master is fixed, it was a weird Travis issue. I'll go ahead and merge 😁

**Pull request merged**